nixos/modules/services/web-apps/akkoma.nix

```diff
@@ -0,0 +1,1031 @@
+    ALTER DATABASE ${escapeSqlId db.database}
+      OWNER TO ${escapeSqlId db.username};
+
+    \connect ${escapeSqlId db.database}
+    CREATE EXTENSION IF NOT EXISTS citext;
+    CREATE EXTENSION IF NOT EXISTS pg_trgm;
+    CREATE EXTENSION IF NOT EXISTS "uuid-ossp";
+  '';
+
+  dbHost = if db ? socket_dir then db.socket_dir
+    else if db ? socket then db.socket
+      else if db ? hostname then db.hostname
+        else null;
+
+  initDbScript = writeShell {
+    name = "akkoma-initdb";
+    runtimeInputs = with pkgs; [ coreutils replace-secret config.services.postgresql.package ];
+    text = ''
+      pgpass="$(mktemp -t pgpass-XXXXXXXXXX.conf)"
+      setupSql="$(mktemp -t setup-XXXXXXXXXX.psql)"
+      trap 'rm -f "$pgpass $setupSql"' EXIT TERM
+
+      ${optionalString (dbHost != null) ''
+        export PGHOST=${escapeShellArg dbHost}
+      ''}
+      export PGUSER=${escapeShellArg cfg.initDb.username}
+      ${optionalString (pgpass != null) ''
+        cat ${escapeShellArg pgpass} >"$pgpass"
+        replace-secret ${escapeShellArgs [
+          (sha256 cfg.initDb.password._secret) cfg.initDb.password._secret ]} "$pgpass"
+        export PGPASSFILE="$pgpass"
+      ''}
+
+      cat ${escapeShellArg setupSql} >"$setupSql"
+      ${optionalString (db ? password) ''
+        replace-secret ${escapeShellArgs [
+         (sha256 db.password._secret) db.password._secret ]} "$setupSql"
+      ''}
+
+      # Create role if non‐existent
+      psql -tAc "SELECT 1 FROM pg_roles
+        WHERE rolname = "${escapeShellArg (escapeSqlStr db.username)} | grep -F -q 1 || \
+        psql -tAc "CREATE ROLE "${escapeShellArg (escapeSqlId db.username)}
+
+      # Create database if non‐existent
+      psql -tAc "SELECT 1 FROM pg_database
+        WHERE datname = "${escapeShellArg (escapeSqlStr db.database)} | grep -F -q 1 || \
+        psql -tAc "CREATE DATABASE "${escapeShellArg (escapeSqlId db.database)}"
+          OWNER "${escapeShellArg (escapeSqlId db.username)}"
+          TEMPLATE template0
+          ENCODING 'utf8'
+          LOCALE 'C'"
+
+      psql -f "$setupSql"
+    '';
+  };
+
+  envWrapper = pkgs.runCommandLocal "akkoma-env" {
+    passAsFile = [ "text" ];
+    text = ''
+      #!${pkgs.runtimeShell}
+      set -e -u -o pipefail
+
+      cd "${cfg.package}"
+
+      RUNTIME_DIRECTORY="''${RUNTIME_DIRECTORY:-/run/akkoma}"
+      AKKOMA_CONFIG_PATH="$RUNTIME_DIRECTORY/config.exs" \
+      ERL_EPMD_ADDRESS="${cfg.dist.address}" \
+      ERL_EPMD_PORT="${toString cfg.dist.epmdPort}" \
+      ERL_FLAGS="${concatStringsSep " " [
+        "-kernel inet_dist_use_interface '${erlAddr cfg.dist.address}'"
+        "-kernel inet_dist_listen_min ${toString cfg.dist.portMin}"
+        "-kernel inet_dist_listen_max ${toString cfg.dist.portMax}"
+      ]}" \
+      RELEASE_COOKIE="$(<"$RUNTIME_DIRECTORY/cookie")" \
+      RELEASE_NAME="akkoma" \
+        exec "${cfg.package}/bin/$(basename "$0")" "$@"
+    '';
+  } ''
+    target="$out/bin/pleroma"
```

Comment:
Doesn't this do the same as `pkgs.writers.writeShellApplication` wrapped in a derivation that adds the second symlink? If so, consider doing that instead, as that allows us to keep this file agnostic of things like shellcheck.

Reply:
That’s a good idea. I’ll change it according to your suggestion.